File: src/lib/ipc/commands/instances.ts
import type { AppInstance } from "../../../types/generated/AppInstance";
import type { DesktopApp } from "../../../types/generated/DesktopApp";
import type { DesktopAppId } from "../../../types/generated/DesktopAppId";

export interface InstanceCommands {
  list_desktop_apps: { args: Record<string, never>; result: DesktopApp[] };
  list_app_instances: { args: { app: DesktopAppId }; result: AppInstance[] };
  create_app_instance: { args: { app: DesktopAppId; name: string }; result: AppInstance };
  start_app_instance: { args: { id: string }; result: AppInstance };
  stop_app_instance: { args: { id: string }; result: AppInstance };
  delete_app_instance: { args: { id: string }; result: void };
}

export type { AppInstance, DesktopApp, DesktopAppId };

File: src-tauri/src/commands/instances.rs
//! Thin adapters for desktop-app multi-instance (create / start / stop / list).

use skillstar_app::instances::{self, AppInstanceDto, DesktopAppDto};
use skillstar_core::infra::error::AppError;

#[tauri::command]
pub fn list_desktop_apps() -> Vec<DesktopAppDto> {
    instances::list_desktop_apps()
}

#[tauri::command]
pub fn list_app_instances(app: String) -> Result<Vec<AppInstanceDto>, AppError> {
    instances::list_instances(&app)
}

#[tauri::command]
pub fn create_app_instance(app: String, name: String) -> Result<AppInstanceDto, AppError> {
    instances::create_instance(&app, name)
}

#[tauri::command]
pub fn start_app_instance(id: String) -> Result<AppInstanceDto, AppError> {
    instances::start_instance(&id)
}

#[tauri::command]
pub fn stop_app_instance(id: String) -> Result<AppInstanceDto, AppError> {
    instances::stop_instance(&id)
}

#[tauri::command]
pub fn delete_app_instance(id: String) -> Result<(), AppError> {
    instances::delete_instance(&id)
}
